Export cmake config package

ClausKlein opened the pull request.

steve-downey committed:
Add verify to build presets

Add the verify headers target to the build presets for each compiler.

steve-downey commented:
I'll try to get to this soon. I'm at the ISO C++ meeting in Poland this week, so terribly busy and 8 hours jet lagged.

ClausKlein committed:
Export cmake config package

ClausKlein committed:
Add find-package-test

ClausKlein committed:
Restructure cmake preset

ClausKlein committed:
Trigger find-packge-test

ClausKlein committed:
Test to find installed package on CI too

ClausKlein committed:
Add missing Debug build type for CI

ClausKlein committed:
Setup that test depends on all

ClausKlein committed:
Use cmake workflow preset on CI

Update README
Bump minimum cmake version

ClausKlein committed:
Use right toolchain for preset name

ClausKlein committed:
It seems clang-17 builds only with -std=c++20 ?

ednolan committed:
Add support for Coveralls to CI

This commit provides support for Coveralls CI by leveraging the
existing Gcov build configuration, adding configuration so that gcovr
generates coverage data in Coveralls JSON format, and adding steps to
the CI build job that are activated only if the matrix entry is
configured for coverage, which generate coverage data and deliver it
to Coveralls.

It also adds a coverage badge to the README.

ednolan committed:
Merge pull request #87 from bemanproject/enolan_coveralls

Add support for Coveralls to CI

ClausKlein committed:
Merge branch 'main' into feature/export-config-package

ClausKlein committed:
Fix merge problems

wusatosi committed:
remove unuseed .cmakg-format

wusatosi committed:
add back shell-check

wusatosi committed:
add back codespell

wusatosi committed:
Revert "add back shell-check"

This reverts commit 7e886f91976d1e2001586c343a6f25805dd37d68.

wusatosi committed:
remove shell check

wusatosi committed:
update pre-commit dependency

camio committed:
Fix typo in CMakePresets.json

Fixes #86.

wusatosi committed:
fix git blame ignore

steve-downey committed:
Merge pull request #90 from wusatosi/blame-fix

Fix git blame ignore for GitHub

wusatosi committed:
Merge pull request #88 from wusatosi/cleanup

Update pre-commit configs

neatudarius committed:
Merge pull request #89 from bemanproject/camio-patch-1

Fix typo in CMakePresets.json

ClausKlein committed:
Merge branch 'main' into feature/export-config-package

ClausKlein committed:
Validate and format json files too

A-QUI15 commented:
Archivos modificado
